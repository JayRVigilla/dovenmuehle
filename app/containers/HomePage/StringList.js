/*
 * StringList
 *
 * creates list item for each string from Server
 * accepts as a prop
 */

import React from 'react';
import PropTypes from 'prop-types';
// import { useSelector } from 'react-redux';
// import { getAllStrings } from '../../ApiCalls';

function StringList({ stringList, madeReq }) {
  StringList.propTypes = {
    stringList: PropTypes.array,
  };

  function renderedList(arr) {
    if (arr === undefined) {
      return (
        <div>
          <p>Array passed to component is undefined</p>
        </div>
      );
    }

    if (arr.length === 0) {
      return (
        <div>
          <p>No strings in list</p>
        </div>
      );
    }
    return (
      <div>
        <ul>
          {arr.map(str => (
            <li key={str.key}>{str.val}</li>
          ))}
        </ul>
      </div>
    );
  }

  function loadingMessage() {
    return (
      <div>
        <p>Loading...</p>
      </div>
    );
  }

  return madeReq === false ? loadingMessage() : renderedList(stringList);
}

export default StringList;
