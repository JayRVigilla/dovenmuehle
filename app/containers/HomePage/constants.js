/*
 * HomePage Constants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const GET_STRINGS = 'dovenmuehle/HomePage/GET_STRINGS';
export const STORE_STRINGS = 'dovenmuehle/HomePage/STORE_STRINGS';
export const GET_STRINGS_ERR = 'dovenmuehle/HomePage/GET_STRINGS_ERR';
