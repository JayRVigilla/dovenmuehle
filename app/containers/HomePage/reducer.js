/*
 *
 * GetString reducer
 *
 */

/* eslint-disable default-case, no-param-reassign */
// import { getAllStrings } from '../../ApiCalls';
const INITIAL_STATE = { strings: [] };

const reducer = (state = INITIAL_STATE, action) => {
  const newState = { ...state };
  switch (action.type) {
    case 'GET_STRINGS_ASYNC':
      newState.strings = action.value;
      break;
  }
  return newState;
};
// end

export default reducer;
