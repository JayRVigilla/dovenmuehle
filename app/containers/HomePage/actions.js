import { GET_STRINGS, STORE_STRINGS, GET_STRINGS_ERR } from './constants';

export function getStringsArray() {
  return {
    type: GET_STRINGS,
  };
}

export function storeStrings(strings) {
  return {
    type: STORE_STRINGS,
    strings,
  };
}

export function errorStrings(err) {
  return {
    type: GET_STRINGS_ERR,
    err,
  };
}
