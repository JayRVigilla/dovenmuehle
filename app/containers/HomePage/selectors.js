import { createSelector } from 'reselect';
import { INITIAL_STATE } from './reducer';

const selectHomePage = state => state.homePage || INITIAL_STATE;

const makeSelectStrings = () =>
  createSelector(
    selectHomePage,
    homePageState => homePageState.strings,
  );

const makeSelectIsLoading = () =>
  createSelector(
    selectHomePage,
    homePageState => homePageState.isLoading,
  );

const makeSelectErr = () =>
  createSelector(
    selectHomePage,
    homePageState => homePageState.err,
  );

export {
  selectHomePage,
  makeSelectStrings,
  makeSelectIsLoading,
  makeSelectErr,
};
