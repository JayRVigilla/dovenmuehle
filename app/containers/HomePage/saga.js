import { put, takeLatest } from 'redux-saga/effects';
import request from 'utils/request';
import { call } from 'file-loader';
import { storeStrings, errorStrings } from '../App/actions';
import { GET_STRINGS } from '../../constants';

export function* getStringsFromAPI() {
  console.log('running getStringsFromAPI()');
  const backendURL = 'http://localhost:3000/data';

  try {
    const { strings } = yield call(request, backendURL);
    yield put(storeStrings, strings);
    console.log(strings);
  } catch (err) {
    yield put(errorStrings, err);
  }
}

export default function* homePageSaga() {
  console.log('running HomePage rootSaga()');
  // should be going to getStringsFromAPI
  yield takeLatest(GET_STRINGS, getStringsFromAPI);
}
