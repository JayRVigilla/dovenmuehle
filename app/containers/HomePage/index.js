/*
 * HomePage
 *
 * found at '/'
 * Holds StringList component, passes props
 * makes axios call and passes strings as prop
 */

import React, { useEffect } from 'react';
import { FormattedMessage } from 'react-intl';
import { compose } from 'redux';
import { connect } from 'react-redux';
import injectSaga from 'utils/injectSaga';
import useInjectReducer from 'utils/injectReducer';
import { createStructuredSelector } from 'reselect';
import saga from './saga';
import messages from './messages';
import StringList from './StringList';
import reducer from './reducer';
import { getStringsArray } from '../App/actions';
import {
  makeSelectStrings,
  makeSelectMadeReq,
  makeSelectErr,
} from './selectors';

const key = 'homePage';

const mapStateToProps = createStructuredSelector({
  madeReq: makeSelectMadeReq(),
  err: makeSelectErr(),
  strings: makeSelectStrings(),
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = useInjectReducer({ key, reducer });

const withSaga = injectSaga({
  key,
  saga,
});

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(function HomePage({ dispatchGetStrings, strings, madeReq }) {
  useEffect(() => dispatchGetStrings(), []);
  console.log('strings from homepage', strings);

  return (
    <div>
      <h1>
        <FormattedMessage {...messages.header} />
      </h1>

      <div>
        <p>Want to add more strings?</p>
        <a href="/add">Click Here</a>
      </div>

      <div>
        <StringList stringList={strings} madeReq={madeReq} />
      </div>
    </div>
  );
});

export function mapDispatchToProps(dispatch) {
  return {
    dispatchGetStrings: evt => {
      if (evt !== undefined && evt.preventDefault) {
        evt.preventDefault();
        dispatch(getStringsArray);
      }
    },
  };
}
