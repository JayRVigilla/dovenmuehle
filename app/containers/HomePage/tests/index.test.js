import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import HomePage from '../index';

describe('<HomePage />', () => {
  beforeAll(() => {});

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <HomePage />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should fetch strings from backend on mount', () => {
    expect().toSomething();
  });

  it('', () => {
    expect().toSomething();
  });
});
