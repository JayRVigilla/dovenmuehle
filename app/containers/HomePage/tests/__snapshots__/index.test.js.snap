// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div>
  <h1>
    <span>
      Write your mystery plot line
    </span>
  </h1>
  <div>
    <p>
      Every good mystery writer knows the secret to success is to work backwards. Read your mystery here.
    </p>
    <a
      href="/add"
    >
      Click Here to add towards the beginning of your mystery
    </a>
  </div>
  <div>
    <div>
      <p>
        Something happened
      </p>
      <p>
        fetch is not defined
      </p>
    </div>
  </div>
</div>
`;
